compiler/next/lib/uast/TupleDecl.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/uast/TupleDecl.h"
+
+#include "chpl/uast/Builder.h"
+
+namespace chpl {
+namespace uast {
+
+bool TupleDecl::assertAcceptableTupleDecl() {
+  int i = 0;
+  for (const auto& elt: children_) {
+    if (i == typeExpressionChildNum_) {
+      if (!elt->isExpression()) {
+        assert(false && "type expression child is not expression"); 
+        return false;
+      }
+    } else if (i == initExpressionChildNum_) {
+      if (!elt->isExpression()) {
+        assert(false && "init expression child is not expression"); 
+        return false;
+      }
+    } else if (!elt->isVariable() && !elt->isTupleDecl()) {
+      assert(false && "other child is not variable or tuple decl");
+      return false;
```

Comment:
This seems unnecessarily complex. What if we have just this:
```C++
if (i == typeExpressionChildNum_)
  assert(elt->isExpression());
else if ......
```

I understand that this way we can't return `false` in these cases. However, I think assertAcceptableTupleDecl() can simply return `void`.